lib/datasource/cargo/index.ts

```diff
@@ -103,8 +107,7 @@ export async function getPkgReleases({
       err.statusCode === 429 ||
       (err.statusCode >= 500 && err.statusCode < 600)
     ) {
-      logger.warn({ lookupName, err }, `cargo crates.io registry failure`);
```

Comment:
My goal is to remove the need for each datasource to log its own warning like this and instead have a consistent log format from the datasource index.

Reply:
We should add the datasource name to the error.

Reply:
The datasource index that catches it already knows the datasource name and lookupName and logs them. Beyond that when it's re-thrown, the special fields are not used.

Reply:
If we *were* to add the additional fields to the Error, it can be done in the index file to save duplicated code within every datasource